packages/xo-server-sdn-controller/src/index.js

```diff
@@ -669,6 +639,60 @@ class SDNController extends EventEmitter {
     }
   }
 
+  async _addHostToPoolNetworks(host) {
+    const xapi = host.$xapi
+
+    const tunnels = filter(xapi.objects.all, { $type: 'tunnel' })
+    for (const tunnel of tunnels) {
+      const accessPIF = await xapi._getOrWaitObject(tunnel.access_PIF)
+      if (accessPIF.host !== host.$ref) {
+        continue
+      }
+
+      const poolNetwork = find(this._poolNetworks, {
+        network: accessPIF.network,
+      })
+      if (poolNetwork == null) {
+        continue
+      }
+
+      if (accessPIF.currently_attached) {
+        continue
+      }
+
+      log.debug(
+        `Pluging PIF: '${accessPIF.device}' for host: '${host.name_label}' on network: '${accessPIF.$network.name_label}'`
+      )
+      try {
+        await xapi.call('PIF.plug', accessPIF.$ref)
+      } catch (error) {
+        log.error(
+          `XAPI error while pluging PIF: '${accessPIF.device}' on host: '${host.name_label}' for network: '${accessPIF.$network.name_label}'`
+        )
```

Comment:
Update your `log.*` in this PR to add contextual data (e.g. here `error`).

Reply:
Casting `error` as `String` may not work correctly.

Why not use the contextual values I told you about? (see [`@xen-orchestra/log`'s readme](https://github.com/vatesfr/xen-orchestra/blob/master/@xen-orchestra/log/README.md).

Reply:
On it, i'm making a new PR to change all the logs with contextual values. I hope it'll make them shorter and easier to read! ^^'